### PR TITLE
Add `--transport` option to port-forward

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
The latest Kubernetes beta supports using websockets for port-forward. We now introduce a similar option to `attach` and `exec` to be able to select the new transport.

See: https://github.com/kubernetes/enhancements/issues/4006

#### Which issue(s) this PR fixes:

Follow-up on https://github.com/kubernetes-sigs/cri-tools/pull/1383

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Added `--transport / -r` flag to `crictl port-forward` and `attach` to select between the "spdy" (default) and "websocket" transport.
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@kubernetes-sigs/cri-tools-maintainers PTAL

File: cmd/crictl/portforward.go [hunk 1]
```diff
@@ -36,6 +37,14 @@ var runtimePortForwardCommand = &cli.Command{
 	Name:      "port-forward",
 	Usage:     "Forward local port to a pod",
 	ArgsUsage: "POD-ID [LOCAL_PORT:]REMOTE_PORT",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:    transportFlag,
+			Aliases: []string{"r"},
```
Comment (kwilczynski) [inline cmd/crictl/portforward.go:43]: Was `r` chosen here because `t` was already taken?

File: cmd/crictl/portforward.go [hunk 2]
```diff
@@ -36,6 +37,14 @@ var runtimePortForwardCommand = &cli.Command{
...
+			Value:   transportSpdy,
+			Usage:   fmt.Sprintf("Transport protocol to be used, must be one of: %s, %s", transportSpdy, transportWebsocket),
```
Comment (kwilczynski) [inline cmd/crictl/portforward.go:45]: To align with the other help formats.

```suggestion
			Usage:   fmt.Sprintf("Transport protocol to use, can be one of: %s|%s", transportSpdy, transportWebsocket),
```

Even closer to the format:

```suggestion
			Usage:   fmt.Sprintf("Transport protocol to use, one of: %s|%s", transportSpdy, transportWebsocket),
```


Also, moved the "must" to "can" as there is a default set, which makes this option optional.

File: cmd/crictl/portforward.go (hunk 1 above)
Comment (saschagrunert) [reply cmd/crictl/portforward.go:43]: Yes, on `exec` and `attach`.

File: cmd/crictl/portforward.go (hunk 2 above)
Comment (saschagrunert) [reply cmd/crictl/portforward.go]: Fixed as suggested :+1:

File: cmd/crictl/portforward.go (hunk 1 above)
Comment (kwilczynski) [reply cmd/crictl/portforward.go:43]: Ah bummer.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Pull request merged**